### PR TITLE
Refactor_sprint5

### PR DESCRIPTION
**Pull request by 3pmile**

### CODE REVIEW

File: src/integer/mat_z/arithmetic/add.rs [hunk 1]
```diff
@@ -104,80 +104,76 @@ mod test_add {
...
     /// testing add_safe
     #[test]
     fn add_safe() {
-        let a: MatZ = MatZ::from_str(&String::from("[[1, 2, 3],[3, 4, 5]]")).unwrap();
-        let b: MatZ = MatZ::from_str(&String::from("[[1, 2, 3],[3, -4, 5]]")).unwrap();
+        let a: MatZ = MatZ::from_str("[[1, 2, 3],[3, 4, 5]]").unwrap();
+        let b: MatZ = MatZ::from_str("[[1, 2, 3],[3, -4, 5]]").unwrap();
         let c = a.add_safe(&b);
-        assert!(c.unwrap() == MatZ::from_str(&String::from("[[2, 4, 6],[6, 0, 10]]")).unwrap());
+        assert_eq!(
```
Comment (Marvin-Beckmann) [inline src/integer/mat_z/arithmetic/add.rs:165]: There are still some assert!( * == *) in arithmetics for PolyOverZ.

**Review by Marvin-Beckmann — CHANGES_REQUESTED** [review]
Loooks good, just two other comments:

- doc comment for power trait has to be fixed
- cargo doc still has a warning: `warning: unused manifest key: bench.0.autodiscovery` can you find out why (potentially ask @MoogSven if he knows)

**Comment by 3pmile** [comment]
@MoogSven why do we need autodiscovery = false in benches? 
It was unused and i removed it for now

**Comment by MoogSven** [comment]
Seems to also work without `autodiscovery`, therefore it can be removed.

**Review by Marvin-Beckmann — APPROVED** [review]

**Pull request merged**